lambda-deployment-orca/src/main/java/com/amazon/aws/spinnaker/plugin/lambda/traffic/LambdaTrafficUpdateVerificationTask.java

```diff
@@ -64,7 +68,38 @@ public TaskResult execute(@NotNull StageExecution stage) {
             return formErrorTaskResult(stage,op.getErrors().getMessage());
         }
 
+        if (!"$WEIGHTED".equals(stage.getContext().get("deploymentStrategy"))) {
+            boolean valid = validateWeights(stage);
+            if (!valid) {
+                formErrorTaskResult(stage, "Could not update weights in time");
+                return TaskResult.builder(ExecutionStatus.TERMINAL).outputs(stage.getOutputs()).build();
+            }
+        }
+
         copyContextToOutput(stage);
         return taskComplete(stage);
     }
+
+    private boolean validateWeights(StageExecution stage) {
+        utils.await(30000);
+        AliasRoutingConfiguration weights = null;
+        long startTime = System.currentTimeMillis();
+        LambdaTrafficUpdateInput inp = utils.getInput(stage, LambdaTrafficUpdateInput.class);
+        boolean status = true;
+        do {
+            utils.await(3000);
+            LambdaDefinition lf = utils.retrieveLambdaFromCache(stage, false);
+            Optional<AliasConfiguration> aliasConfiguration = lf.getAliasConfigurations().stream().filter(al -> al.getName().equals(inp.getAliasName())).findFirst();
+
+            if (aliasConfiguration.isPresent()) {
+                Optional<AliasRoutingConfiguration> opt = Optional.ofNullable(aliasConfiguration.get().getRoutingConfig());
+                weights = opt.orElse(null);
+            }
+            if ((System.currentTimeMillis()-startTime)>240000) {
+                logger.warn("validateWeights function is taking to much time");
```

Comment:
```suggestion
                logger.warn("validateWeights function is taking too much time");
```
and maybe include the actual time and the limit in the log message?